src/child/actions/window.js

```diff
@@ -71,15 +78,22 @@ export function resizeToCompact() {
     };
 }
 
-export function resizeToDefault() {
+export function resizeToPrevious(dimensions, isMaximised) {
     return dispatch => {
         dispatch(resizing());
-        const defaultWindowDimensions = configService.getDefaultWindowDimensions();
         fin.desktop.Window.getCurrent().resizeTo(
-            defaultWindowDimensions[0],
-            defaultWindowDimensions[1],
+            dimensions[0],
+            dimensions[1],
             'top-right',
-            () => dispatch(expand()),
+            () => {
+                dispatch(expand());
+                if (isMaximised) {
```

Comment:
If the window is going to be maximised, does `resizeTo()` still need to be called, or is `maximize()` enough?

I think this raises an interesting point too - should `isMaximised` be set to `false` when `isCompact` is `true` - it seems contradictory that `isCompact` and `isMaximised` can both be `true` simultaneously?